truffle/src/main/java/org/jruby/truffle/nodes/core/BigDecimalNodes.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright (c) 2013, 2015 Oracle and/or its affiliates. All rights reserved. This
+ * code is released under a tri EPL/GPL/LGPL license. You can use it,
+ * redistribute it and/or modify it under the terms of the:
+ *
+ * Eclipse Public License version 1.0
+ * GNU General Public License version 2
+ * GNU Lesser General Public License version 2.1
+ */
+package org.jruby.truffle.nodes.core;
+
+import com.oracle.truffle.api.CompilerDirectives;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.nodes.Node;
+import com.oracle.truffle.api.object.*;
+import com.oracle.truffle.api.source.SourceSection;
+
+import org.jruby.truffle.nodes.objects.Allocator;
+import org.jruby.truffle.runtime.RubyContext;
+import org.jruby.truffle.runtime.core.*;
+
+import java.math.BigDecimal;
+import java.util.EnumSet;
+
+@CoreClass(name = "Truffle::BigDecimal")
+public abstract class BigDecimalNodes {
+
+    private static final HiddenKey VALUE_IDENTIFIER = new HiddenKey("value");
+    private static final DynamicObjectFactory BIG_DECIMAL_FACTORY;
+    public static final Property VALUE_PROPERTY;
+
+    static {
+        Shape.Allocator allocator = RubyBasicObject.LAYOUT.createAllocator();
+        VALUE_PROPERTY = Property.create(
+                VALUE_IDENTIFIER,
+                allocator.locationForType(BigDecimal.class, EnumSet.of(LocationModifier.NonNull, LocationModifier.Final)),
+                0);
+        Shape shape = RubyBasicObject.EMPTY_SHAPE.addProperty(VALUE_PROPERTY);
+        BIG_DECIMAL_FACTORY = shape.createFactory();
+    }
+
+    // TODO (pitr 15-May-2015) figure out where to put RubyBigDecimal, or remove completely
```

Comment:
Better to remove it sooner than later. I think not supporting subclasses is an acceptable drawback until we properly check the layout or shape.